src/python/pants/source/source_root.py

```diff
@@ -253,6 +254,22 @@ def register_options(cls, register):
 
   @memoized_method
   def get_source_roots(self):
+    deprecated_conditional(
+      lambda: self.options.is_default("unmatched"),
+      removal_version="1.28.0.dev0",
+      entity_description="Defaulting to `--source-unmatched=create`",
+      hint_message=(
+        "`--source-unmatched=create` is an unsafe option. It means that when no source root can be "
+        "found for a file, such as `src/python` or `tests/java`, Pants will try to automatically "
+        "create an implicit source root. Instead, when Pants does not already know about the "
+        "source root, you should explicitly register the source root in your `pants.ini`. "
+        "Run `./pants help-advanced source` for more information.\n\nIn 1.28.0.dev0, Pants will "
+        "change the default to `--source-unmatched=fail`, and in 1.30.0.dev0, the option will be "
+        "removed and Pants will always fail.\n\nTo prepare for this change, please explicitly set "
```

Comment:
Oh, whoa. Missed this part.

Totally fine with changing the default, but we can't remove `create`, unfortunately... it's very widely used in our repository for configuration that needs to be located at the "bottom" of a source root.

Reply:
No longer trying to remove `create`. This PR now only deprecates the default to change to the sane value.